src/Microsoft.DotNet.ProjectJsonMigration/Rules/MigrateBuildOptionsRule.cs

```diff
@@ -23,7 +23,7 @@ private AddPropertyTransform<CommonCompilerOptions>[] EmitEntryPointTransforms
                 new AddPropertyTransform<CommonCompilerOptions>("OutputType", "Exe",
                     compilerOptions => compilerOptions.EmitEntryPoint != null && compilerOptions.EmitEntryPoint.Value),
                 new AddPropertyTransform<CommonCompilerOptions>("OutputType", "Library",
-                    compilerOptions => compilerOptions.EmitEntryPoint == null || !compilerOptions.EmitEntryPoint.Value)
+                    compilerOptions => compilerOptions.EmitEntryPoint != null && !compilerOptions.EmitEntryPoint.Value)
```

Comment:
Yes no entry point defaults to the default. 

This was breaking between different configurations because emitEntrypoint was null and Library was being output.